django-rgd-imagery/tests/test_rest.py

```diff
@@ -83,3 +91,33 @@ def test_create_and_download_cog(admin_api_client, geotiff_image_entry):
     pk = cog.pk
     response = admin_api_client.get(f'/api/image_process/{pk}')
     assert response.data
+
+
+@pytest.mark.django_db(transaction=True)
+def test_tiles_endpoint_with_signature(admin_api_client, live_server):
+    image = factories.ImageFactory(
+        file__file__filename='paris_france_10.tiff',
+        file__file__from_path=datastore.fetch('paris_france_10.tiff'),
+    )
+    # Set the TTL to something short
+    prior = getattr(settings, 'RGD_SIGNED_URL_TTL', None)
+    settings.RGD_SIGNED_URL_TTL = 3  # seconds
+    # Generate a signature
+    response = admin_api_client.post('/api/signature')
+    params = response.data
+    # 15/16618/11252 - paris_france_10.tiff
+    url = f'{live_server.url}/api/image_process/imagery/{image.pk}/tiles/15/16618/11252.png?projection=EPSG:3857'
+    for k, v in params.items():
+        url += f'&{k}={v}'
+    # Use a client without authententication
+    client = RequestsClient()
+    response = client.get(url)
+    response.raise_for_status()
+    # Let the signature expire and assert that we get an accessed denied error
+    time.sleep(3)
+    response = client.get(url)
+    with pytest.raises(requests.HTTPError):
+        response.raise_for_status()
+    assert response.status_code == 401
+    # Reset the TTL now that test is over
+    settings.RGD_SIGNED_URL_TTL = prior
```

Comment:
Would it be worth using a [`settings` fixture](https://pytest-django.readthedocs.io/en/latest/configuring_django.html#overriding-individual-settings) here? Then you don't have to worry about resetting the TTL at the end of the test, and we wouldn't need to remember to do this for future tests written around this Django setting.

Reply:
Thanks! I didn't know about this fixture. I updated other places we were using the settings in tests